appserver/deployment/dol/src/main/java/com/sun/enterprise/deployment/util/ResourceValidator.java

```diff
@@ -841,12 +860,28 @@ private void validateJNDIRefs(AppResource resource, JNDINamespace namespace) {
         if (isResourceInDomainXML(jndiName) || isDefaultResource(jndiName)) {
             return;
         }
+       
 
         // Managed Bean & EJB portable JNDI names
         if (jndiName.startsWith(ResourceConstants.JAVA_MODULE_SCOPE_PREFIX) || jndiName.startsWith(ResourceConstants.JAVA_APP_SCOPE_PREFIX)) {
+            // first look with the name
+            if (namespace.find(jndiName, env)) {
+                return;
+            }
+            
             String newName = convertModuleOrAppJNDIName(jndiName, resource.getEnv());
             if (namespace.find(newName, env)) {
                 return;
+            } else {
```

Comment:
Just a comment, but since the if statement always returns, using else is not needed. The code in the else's body can just be put below the if.